src/lib/define-dynamic-block.js

```diff
@@ -5,6 +5,15 @@ import BlockType from 'scratch-vm/src/extension-support/block-type';
 import ContextMenuContext from 'scratch-vm/src/extension-support/context-menu-context';
 import log from './log.js';
 
+const updateBlockOnWorkspace = (ScratchBlocks, workspace, blockId, blockInfo) => {
+    const block = workspace.getBlockById(blockId);
+    // TODO this seems hacky but is needed for how defineDynamicBlock currently works.
+    block.needsBlockInfoUpdate = true;
+    block.blockInfoText = JSON.stringify(blockInfo);
+    const mutation = block.mutationToDom();
+    block.domToMutation(mutation);
```

Comment:
I think I'd prefer these 4 lines to be in a method on the block so that we can treat `blockInfoText` as somewhat private to the block. I believe it would work to just add another method to the object returned by `defineDynamicBlock`.